server/blob/blob-union/src/main/java/org/apache/james/blob/union/UnionBlobStore.java

```diff
@@ -137,6 +137,18 @@ public InputStream read(BucketName bucketName, BlobId blobId) {
         }
     }
 
+    @Override
+    public Mono<Void> deleteBucket(BucketName bucketName) {
+        try {
+            return Mono.defer(() -> currentBlobStore.deleteBucket(bucketName))
+                .onErrorResume(this::logAndReturnEmpty)
+                .switchIfEmpty(legacyBlobStore.deleteBucket(bucketName));
```

Comment:
I do believe here we want to delete the bucket **in both** blobStores - if any of them fail then we should return an error.